pkg/brokerapi/fake/fake.go

```diff
@@ -19,16 +19,34 @@ package fake
 import (
 	"fmt"
 
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
 	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
 	uuid "github.com/satori/go.uuid"
 )
 
 // Client implements a fake broker API client, useful for unit testing. None of the methods on
 // the client are concurrency-safe
 type Client struct {
-	CatalogClient
-	InstanceClient
-	BindingClient
+	*CatalogClient
+	*InstanceClient
+	*BindingClient
+}
+
+// NewClientFunc returns a function suitable for creating a new BrokerClient from a given
+// Broker object. The returned function is suitable for passing as a callback to code that
+// needs to create clients on-demand
+func NewClientFunc(
```

Comment:
I'm not sure I completely understand why there's so much indirection in this particular spot. Why do we have a function that returns a function that returns a new client? (Instead of _just_ having function(s) that return a new client?) I'm sure you have a reason, but I can't see it atm.